### ISSUE TITLE
Cannot reproduce the paper's results on BVCC test set

### ISSUE THREAD

**Issue by unilight** [soumimaiti/speechlmscore_tool#4]
In case anyone is trying to use this repo to reproduce the BVCC test set results on the paper,
my results are:

[UTT][ MSE = 19.502 | LCC = 0.437 | SRCC = 0.437 | KTAU = 0.304 ]
[SYS][ MSE = 20.428 | LCC = 0.5795 | SRCC = 0.5851  | KTAU = 0.414 ]

I can only get a system-level SRCC of 0.5851.

If we look at Table 2 in the paper (https://arxiv.org/pdf/2212.04559), I don't know which row the pre-trained model corresponds to, but all rows have a system-level SRCC over 0.7.

Not really sure what is missing. For me, one thing that confused me was that I didn't know which HuBERT layer should I use. Apparently, the pre-trained k-means model in this repo is simply the one from the fairseq GSLM repo, so I assume it's the 6th layer. All other layers I tried gave me lower SRCC scores.

@ecooper7 also got the same results.

**Comment by sw005320** [comment]
Thanks for your report, @unilight.

@soumimaiti and @Takaaki-Saeki, can you respond to this question?

**Comment by Takaaki-Saeki** [comment]
Hi @unilight, thanks for sharing it. 

I'm not sure about the scoring script used for the experiment in the paper.
@soumimaiti, can you please respond?

**Comment by soumimaiti** [comment]
@unilight We trim/pad audio for longer/shorter silences in the beginning or in the end. You can use this script https://github.com/soumimaiti/speechlmscore_tool/blob/main/trim_pad_audio.sh 

You can also use https://github.com/soumimaiti/speechlmscore_tool/blob/main/run_bvcc.sh to get results for BVCC results. We use layer 3.

**Comment by ecooper7** [comment]
Thanks, @soumimaiti!  I was able to get results closer to the ones in the paper by running the trim/pad script on the audio first.

Just to confirm -- does the pretrained model correspond to the one called `SpeechLMScore (LSTM)+rep` in Table 2 of the paper?  Were the speechLMs also trained on trimmed/padded audio?

**Comment by soumimaiti** [comment]
@unilight  
1) Yes, pretrained model correspond to SpeechLMScore (LSTM)+rep in Table 2 
2) speechLMs were trained with Librilight segments, which has very small audio silence in the beginning/end. To match, we used trim/pad to approximate small silence.
